### PR TITLE
Olaf/check negative

### PR DESCRIPTION
**Pull request by ocke**
:v:

/cc @zendesk/vegemite

### Description

After some more negative reports, I added a few checks. Doesn't resolve the problem though.

### References
* JIRA: 

### Risks
* None

### CODE REVIEW

**Comment by tobico** [comment]
:+1:

**Comment by svizzari** [comment]
:+1:

File: app.js [hunk 1]
```diff
@@ -382,6 +382,11 @@
     },
 
     updateTime: function(time) {
+      if (time < 0) {
+        this.resetElapsedTime();
+        time = 0;
+        console.error(this.I18n.t('errors.negative_time'));
```
Comment (adammw) [inline app.js]: Do we need to translate a console error? Maybe use services.notify if the user should see it, or just log in english if they shouldn't?

File: app.js [hunk 2]
```diff
@@ -363,7 +363,7 @@
       if (typeof time !== "undefined") {
         this.realElapsedTime = time * 1000;
       }
-      return (this.realElapsedTime / 1000) | 0;
+      return Math.floor(this.realElapsedTime / 1000);
```
Comment (adammw) [inline app.js]: So now we truncate the decimal places? How does this help the issue or is it separate?

Comment (ocke) [inline app.js]: From crazy to no crazy.

**Comment by adammw** [comment]
:+1: but I still have reservations about the error message.

**Comment by ocke** [comment]
FWIW I reverted this code because sometimes people want to add negative numbers
https://github.com/zendesk/timetracking_app/commit/1c51f5270b1fdd6672af8c20ea2318075fedd023

**Pull request merged**